src/core/net/dns_client.hpp

```diff
@@ -136,11 +140,22 @@ class Client : public InstanceLocator, private NonCopyable
         enum Nat64Mode : uint8_t
         {
             kNat64Unspecified = OT_DNS_NAT64_UNSPECIFIED, ///< NAT64 mode is not specified. Use default NAT64 mode.
-            kNat64Allow       = OT_DNS_NAT64_ALLOW,       ///< Allow NAT64 address translation
+            kNat64Allow       = OT_DNS_NAT64_ALLOW,       ///< Allow NAT64 address translation.
             kNat64Disallow    = OT_DNS_NAT64_DISALLOW,    ///< Disallow NAT64 address translation.
         };
 #endif
 
+        /**
+         * This enumeration type represents the DNS transport protocol selection.
+         *
+         */
+        enum TransportProto : uint8_t
+        {
+            kDnsUnspecifiedTransport = OT_DNS_UNSPECIFIED_TRANSPORT, /// Dns transport is unspecified.
+            kDnsUdpTransport         = OT_DNS_UDP_TRANSPORT,         /// Dns query should be sent via UDP.
+            kDnsTcpTransport         = OT_DNS_TCP_TRANSPORT,         /// Dns query should be sent via TCP.
```

Comment:
nit: use common prefix?
```suggestion
            kDnsTransportUnspecified = OT_DNS_TRANSPORT_UNSPECIFIED, /// Dns transport is unspecified.
            kDnsTransportUdp         = OT_DNS_TRANSPORT_UDP,         /// Dns query should be sent via UDP.
            kDnsTransportTcp         = OT_DNS_TRANSPORT_TCP,         /// Dns query should be sent via TCP.
```

Reply:
fixed